### PR TITLE
add rna merge scripts

### PR DESCRIPTION
**Pull request by zhangb1**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- add merge_fusion  and merge_rsem script for the data-assembly 
- the scripts are are orginal written by @kgaonkar6 from here:  https://github.com/d3b-center/ngs_assembly_rna/
- made some changes in the inputs files and remove the data download steps

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Environment: R 
* Test files:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have committed any related changes to the PR

### CODE REVIEW

**Comment by zhangb1** [comment]
testing locally:

```
Rscript 01-create-and-add-fusion-files.R --input_arriba BS_FKASA3MN.arriba_formatted.annotated.tsv --input_star_fusion BS_FKASA3MN.starfusion_formatted.tsv --outdir ./ --mergefiles TRUE --old_arriba pbta-fusion-arriba.tsv.gz --old_starfusion pbta-fusion-starfusion.tsv.gz --outname_prefix pbta
```

```
 Rscript 00-create-and-add-rsem-gene-files.R --input_rsem 806668be-e3a2-4ea3-90fb-f67eba78c7b3.rsem.genes.results.gz --outdir ./ --mergefiles TRUE --old_rsem_count pbta-gene-counts-rsem-expected_count.stranded.rds --old_rsem_fpkm pbta-gene-expression-rsem-fpkm.stranded.rds --old_rsem_tpm pbta-gene-expression-rsem-tpm.stranded.rds --outname_prefix pbta --library stranded --biospecimens_id BS_FKASA3MN
```

**Comment by yuankunzhu** [comment]
@zhangb1 i wonder you could also bring those PR to your Tuesday code review meeting. so the @d3b-center/bix-ops team can really use that to do some real code review instead 😄

File: scripts/merge_fusion/01-create-and-add-fusion-files.R [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# Author: Krutika Gaonkar
+# Merge step from Komal Rathi's code in OpenPBTA
+# Date: 05/10/2020
+# Function:
+# merges new RSEM files into previous version of rsem RDS object
+
```
Comment (kgaonkar6) [inline scripts/merge_fusion/01-create-and-add-fusion-files.R:6]: I'd update the Description  here which looks like I had just copied over from other scripts. 🤦‍♀️

```suggestion
# merges new arriba and starfusion files into previous released version of merged tsv files

```

File: scripts/merge_fusion/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Merge StarFusion and Arriba 
+
+
+### Description
+
+- The folder scripts are orginal written by @kgaonkar6 from here: https://github.com/d3b-center/ngs_assembly_rna/tree/main/analyses/merge_fusion
```
Comment (kgaonkar6) [inline scripts/merge_fusion/README.md:6]: Maybe? 
```suggestion
# Merge StarFusion and Arriba 


### Description

- Modified from [here](https://github.com/d3b-center/ngs_assembly_rna/tree/main/analyses/merge_fusion) to use within cwl workflow
```

File: scripts/merge_rsem/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Merge rsem files
+
+### Description
+
+- The folder scripts are orginal written by @kgaonkar6 from here: https://github.com/d3b-center/ngs_assembly_rna/tree/main/analyses/merge_rsem
```
Comment (kgaonkar6) [inline scripts/merge_rsem/README.md:5]:
```suggestion
- Modified from [here](https://github.com/d3b-center/ngs_assembly_rna/tree/main/analyses/merge_rsem) to use within cwl workflow
```

**Review by kgaonkar6 — COMMENTED** [review]
Looks good to me, just had some minor suggestion for the description sections.

**Comment by zhangb1** [comment]
[test_files.tar.gz](https://github.com/d3b-center/d3b-bixu-data-assembly/files/6266451/test_files.tar.gz)

test file upload here :

old merged files can be download from  s3://d3b-bix-dev-data-bucket/hgg-dmg-integration/20210311-data/

@HuangXiaoyan0106 @sickler-alex

**Comment by HuangXiaoyan0106** [comment]
1. `01-create-and-add-fusion-files.R` looks good to me, I tested it locally. Just curious, old files (`--old_arriba`, `--old_starfusion `) are in "tsv" format, why the output files are in ".gz" format? If i'm not wrong, now they need to be decompressed before exporting to the S3 bucket.
2. For `00-create-and-add-rsem-gene-files.R`,  I downloaded input files from `s3://d3b-bix-dev-data-bucket/hgg-dmg-integration/20210311-data/`.  But get error message:
```
$ Rscript 00-create-and-add-rsem-gene-files.R --input_rsem test_files/181c9d85-9447-4cc2-a2d3-e932b3eb63d9.rsem.genes.results.gz --outdir ./ --mergefiles TRUE --old_rsem_count pbta-gene-counts-rsem-expected_count.stranded.rds --old_rsem_fpkm pbta-gene-expression-rsem-fpkm.stranded.rds --old_rsem_tpm pbta-gene-expression-rsem-tpm.stranded.rds --outname_prefix pbta --library stranded --biospecimens_id BS_FKASA3MN

[1] "test_files/181c9d85-9447-4cc2-a2d3-e932b3eb63d9.rsem.genes.results.gz"
[1] "adding "
Error: `x` and `y` must share the same src, set `copy` = TRUE (may be slow).
Backtrace:
     █
  1. └─`%>%`(...)
  2.   ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
  3.   └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  4.     └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  5.       └─`_fseq`(`_lhs`)
  6.         └─magrittr::freduce(value, `_function_list`)
  7.           ├─base::withVisible(function_list[[k]](value))
  8.           └─function_list[[k]](value)
  9.             ├─dplyr::left_join(...)
 10.             └─dplyr:::left_join.data.frame(...)
 11.               └─dplyr::auto_copy(x, y, copy = copy)
 12.                 └─dplyr:::glubort(...)
Execution halted
```

**Comment by HuangXiaoyan0106** [comment]
Now looks good to me. Will approve.

**Review by HuangXiaoyan0106 — APPROVED** [review]

**Review by kgaonkar6 — APPROVED** [review]

File: scripts/merge_rsem/00-create-and-add-rsem-gene-files.R [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+old_rsem_fpkm <- opt$old_rsem_fpkm
+old_rsem_tpm <- opt$old_rsem_tpm
+old_rsem_count <- opt$old_rsem_count
+outname_prefix <- opt$outname_prefix
+mergefiles <- opt$mergefiles
+library <- opt$library
+bs_id <-opt$biospecimens_id
+
+
+if (mergefiles) {
```
Comment (sickler-alex) [inline scripts/merge_rsem/00-create-and-add-rsem-gene-files.R:66]: If everything but reading the inputs is within the if mergefiles block, why would you want to run the script but not merge the files? You could take out the option mergefiles, remove the if statement, and only run the code when the files need to be merged.

**Review by sickler-alex — CHANGES_REQUESTED** [review]

**Pull request merged**